resilience4j-timelimiter/src/main/java/io/github/resilience4j/timelimiter/internal/TimeLimiterImpl.java

```diff
@@ -55,6 +55,37 @@ public TimeLimiterImpl(String name, TimeLimiterConfig timeLimiterConfig) {
         };
     }
 
+    @Override
+    public <T, F extends CompletionStage<T>> Supplier<CompletionStage<T>> decorateCompletionStage(Supplier<F> supplier) {
+        CompletableFuture<T> future = supplier.get().toCompletableFuture();
```

Comment:
Line 60 already executes the Supplier (which returns a CompletionStage).
The Supplier should not be invoked in `decorateCompletionStage`. But instead a decorated Supplier should be returned.

You can have a look at `CircuitBreaker.decorateCompletionStage` and check where `supplier.get()` is invoked.